contents/stats_vis.md

```diff
@@ -1,7 +1,181 @@
+For the `xticks`' labels we used the `levels` function from `CategoricalArrays.jl` that returns the categorical levels from our `name` variable in the same order as the integer codes.
+Finally, for the `x` vector inside Makie's `boxplot` function, we wrap the `name` variable with the `levelcode` function, also from `CategoricalArrays.jl`, which returns the underlying integer codes from our categorical variable `name`.
+We do this because Makie's `boxplot` only accepts a vector of `Int`s as inputs for the `x` argument.
+Here is the code:
+
+```jl
+s = """
+    CairoMakie.activate!() # hide
+    label = "boxplot" # hide
+    caption = "Box Plot" # hide
+    df = more_grades()
+    transform!(df, :name => categorical; renamecols=false)
+    fig = Figure(; resolution=(600, 400))
+    ax = Axis(fig[1, 1]; xticks = (1:4, levels(df.name)))
+    boxplot!(ax, levelcode.(df.name), df.grade)
+    Options(current_figure(); filename=label, caption, label) # hide
+    """
+sco(s)
+```
+
+The default IQR range for the whiskers in `Makie.jl` is 1.5.
+However, sometimes we see the whiskers either with a different IQR range or with a small vertical bar to better visualize the whiskers tips.
+We can control both of those with the `range` (default `1.5`) and `whiskerwidth` (default `0.0`) arguments:
+
+```jl
+s = """
+    CairoMakie.activate!() # hide
+    label = "boxplot_custom" # hide
+    caption = "Box Plot with different IQR and Whiskers Vertical Bars" # hide
+    df = more_grades()
+    transform!(df, :name => categorical; renamecols=false)
+    fig = Figure(; resolution=(600, 400))
+    ax = Axis(fig[1, 1]; xticks = (1:4, levels(df.name)))
+    boxplot!(ax, levelcode.(df.name), df.grade; range=2.0, whiskerwidth=0.5)
+    Options(current_figure(); filename=label, caption, label) # hide
+    """
+sco(s)
+```
+
+Box plots can also flag anything outside the whiskers as outliers.
+By default, these observations are not shown in `Makie.jl` but you can control it with the `show_outliers` argument:
+
+```jl
+s = """
+    CairoMakie.activate!() # hide
+    label = "boxplot_outliers" # hide
+    caption = "Box Plot with Outliers" # hide
+    df = more_grades()
+    transform!(df, :name => categorical; renamecols=false)
+    fig = Figure(; resolution=(600, 400))
+    ax = Axis(fig[1, 1]; xticks = (1:4, levels(df.name)))
+    boxplot!(ax, levelcode.(df.name), df.grade; range=0.5, show_outliers=true)
+    Options(current_figure(); filename=label, caption, label) # hide
+    """
+sco(s)
+```
+
+As you can see, **box plots are a useful way to visualize data with robust central tendencies and dispersion measures to outliers**.
+
 ### Density Plots {#sec:stats_vis_densityplots}
 
+Box plot limit us just to summary statistics like median, quartiles and IQRs.
+Often we want to see the full distribution of our data.
```

Comment:
```suggestion
Often, we want to get a clearer picture of the distribution of our data.
```
Not the best sentence that I'm suggestion here. I'm trying to convey that density plots (kernel density estimators) are approximations and don't show "the full distribution".

Reply:
Ok, I changed this in the new commit f6c9526

Reply:
Yes, great :+1: thanks